django/core/checks/urls.py

```diff
@@ -28,6 +28,30 @@ def check_resolver(resolver):
         return []
 
 
+@register(Tags.urls)
+def check_url_namespaces_unique(app_configs, **kwargs):
+    """
+    Checks that all URL namespaces used in applications are unique.
+    """
+    from django.urls import get_resolver
+    resolver = get_resolver()
+
+    custom_namespaces = [url.namespace for url in resolver.url_patterns
+                         if getattr(url, 'namespace', None) is not None and url.namespace != 'app']
+
+    included_resolvers = [urlconf for url in resolver.url_patterns for urlconf in getattr(url, 'urlconf_name', [])]
+    included_custom_namespaces = [r.namespace for r in included_resolvers
+                                  if getattr(r, 'namespace', None) is not None and r.namespace != 'app']
+    all_namespaces = custom_namespaces + included_custom_namespaces
+    for namespace in all_namespaces:
+        if all_namespaces.count(namespace) > 1:
+            return [Error(
```

Comment:
I think a `Warning` is more appropriate here, something like "URL namespace {} is not unique, you may not be able to reverse all URLs in this namespace". Errors prevent management commands from running, which is a bit severe for this case.